src/jetson/main.py

```diff
@@ -122,9 +135,66 @@ def detect(self,
                     kp_y = detections[i, 4 + k * 2 + 1] * img.shape[0]
 
                 bboxes.append((xmin, ymin, xmax, ymax))
+
+            return bboxes
+
+
+        elif (self.model_name == 'retinaface'):
+            resize = 1     #Set resize factor
+            img = cv2.resize(image, (int(image.shape[1]*resize), int(image.shape[0]*resize))).astype(np.float32)
+            img_h, img_w = img.shape[0], img.shape[1]
+            scale = torch.Tensor([img.shape[1], img.shape[0], img.shape[1], img.shape[0]])
+            img -= (104, 117, 123)
+            img = img.transpose(2, 0, 1)
+            img = torch.from_numpy(img).unsqueeze(0)
+            img = img.to(self.device)
+            scale = scale.to(self.device)
+
+            loc, conf, landms = self.net(img)  # forward pass
+            priorbox = PriorBox(cfg, image_size=(img_h, img_w))
+            priors = priorbox.forward()
+            priors = priors.to(self.device)
+            prior_data = priors.data
+            boxes = decode(loc.data.squeeze(0), prior_data, cfg['variance'])
+            boxes = boxes * scale / resize
+            boxes = boxes.cpu().numpy()
+            scores = conf.squeeze(0).data.cpu().numpy()[:, 1]
+            scale1 = torch.Tensor([img.shape[3], img.shape[2], img.shape[3], img.shape[2],
+                                   img.shape[3], img.shape[2], img.shape[3], img.shape[2],
+                                   img.shape[3], img.shape[2]])
+            scale1 = scale1.to(self.device)
+
+            # ignore low scores
+            inds = np.where(scores > self.detection_threshold)[0]
+            boxes = boxes[inds]
+            scores = scores[inds]
+
+            # keep top-K before NMS
+            top_k = 1000
+            order = scores.argsort()[::-1][:top_k]
+            boxes = boxes[order]
+            scores = scores[order]
+
+            # do NMS
+            nms_thresh = 0.3
+            dets = np.hstack((boxes, scores[:, np.newaxis])).astype(np.float32, copy=False)
+            keep = nms_numpy(dets, nms_thresh)
+            dets = dets[keep, :]
+
+            # keep top-K faster NMS
+            keep_top_k = 750
+            dets = dets[:keep_top_k, :]
+
+            bboxes = []
+            for det in dets:
+                bboxes.append(tuple(dets[0][0:4]))
```

Comment:
can we break up this logic into functions? I realize we're probably not going to need to reuse this code but by leaving it this long its very hard to read and therefore will be harder to maintain